### PR TITLE
[Docs] Convert more doc snippets

### PR DESCRIPTION
**Pull request by tlrx**
This commit converts some remaining doc snippets so that they are now testable. It caught quite many wrong or missing fields.

### CODE REVIEW

File: docs/reference/cluster/stats.asciidoc [hunk 1]
```diff
@@ -188,8 +193,19 @@ Will return, for example:
...
+         "http_types" : {
+            "netty4" : 1
+         }
+      }
    }
 }
 --------------------------------------------------
+// TESTRESPONSE[s/"plugins": \[[^\]]*\]/"plugins": $body.$_path/]
+// TESTRESPONSE[s/: (\-)?[0-9]+/: $body.$_path/]
+// TESTRESPONSE[s/: "[^"]*"/: $body.$_path/]
```
Comment (tlrx) [inline docs/reference/cluster/stats.asciidoc]: This replace all the fields values by the expected ones in the test. We don't care about the field values, but we need to test the field names.

File: docs/reference/indices/recovery.asciidoc [hunk 2]
```diff
@@ -20,6 +20,10 @@ To see cluster-wide recovery status simply leave out the index names.
 GET /_recovery?human
 --------------------------------------------------
 // CONSOLE
+// TEST[s/^/POST \/_snapshot\/my_repository\/snap_1\/_restore\?wait_for_completion=true\n/]
+// TEST[s/^/DELETE index1\n/]
+// TEST[s/^/PUT \/_snapshot\/my_repository\/snap_1\?wait_for_completion=true\n/]
+// TEST[s/^/PUT \/_snapshot\/my_repository\n{"type": "fs","settings": {"location": "recovery_asciidoc" }}\n/]
```
Comment (tlrx) [inline docs/reference/indices/recovery.asciidoc]: I'm not fully sure of this one. Note: it must be read in reverse order

File: docs/reference/cluster/stats.asciidoc (hunk 1 above)
Comment (tlrx) [inline docs/reference/cluster/stats.asciidoc]: Also, I wasn't able to test this one on Windows and I suspect it fails due to the open file descriptor field.

File: docs/reference/cluster/nodes-info.asciidoc [hunk 3]
```diff
@@ -149,11 +171,41 @@ The result will look similar to:
           "classname": "org.elasticsearch.ingest.useragent.IngestUserAgentPlugin",
           "has_native_controller": false
         }
+      ],
+      "modules": [
+        {
+          "name": "lang-painless",
+          "version": "7.0.0-alpha1",
```
Comment (jpountz) [inline docs/reference/cluster/nodes-info.asciidoc]: let's use {version} like below?

File: docs/reference/cluster/stats.asciidoc (hunk 1 above)
Comment (jpountz) [inline docs/reference/cluster/stats.asciidoc]: can you add a comment about what itdoes in the docs file?

File: docs/reference/indices/recovery.asciidoc (hunk 2 above)
Comment (jpountz) [inline docs/reference/indices/recovery.asciidoc]: maybe you could have a commented snippet (in the asciidoc sense) that does the setup and here just do a TEST[continued]? We do that sometimes to test responses without having to actually put the response in the docs. I'm wondering whether the same trick can be used to quick setups too.

**Review by jpountz — APPROVED** [review]
:heart:

File: docs/reference/cluster/stats.asciidoc (hunk 1 above)
Comment (tlrx) [inline docs/reference/cluster/stats.asciidoc]: Sure

File: docs/reference/cluster/nodes-info.asciidoc (hunk 3 above)
Comment (tlrx) [inline docs/reference/cluster/nodes-info.asciidoc]: Good catch, thanks

File: docs/reference/indices/recovery.asciidoc (hunk 2 above)
Comment (tlrx) [inline docs/reference/indices/recovery.asciidoc]: Good idea, I just forgot that it was possible...

File: docs/reference/cluster/stats.asciidoc (hunk 1 above)
Comment (tlrx) [inline docs/reference/cluster/stats.asciidoc]: It works on windows \o/

**Comment by tlrx** [comment]
Thanks @jpountz

**Comment by nik9000** [comment]
❤️

**Pull request merged**